tools/pipelines/templates/build-npm-package.yml

```diff
@@ -261,22 +270,43 @@ stages:
               targetType: 'inline'
               workingDirectory: ${{ parameters.buildDirectory }}
               script: |
-                mkdir $(Build.ArtifactStagingDirectory)/pack/
+                mkdir $(Build.ArtifactStagingDirectory)/pack-scoped/
+                if [[ "${{ variables.publishNonScopedPackages }}" == "True" ]]; then
+                  mkdir $(Build.ArtifactStagingDirectory)/pack-non-scoped/
+                fi
                 if [ -f "lerna.json" ]; then
                   npx lerna exec --no-private --no-sort -- npm pack --unsafe-perm
-                  npx lerna exec --no-private --no-sort --parallel -- mv -t $(Build.ArtifactStagingDirectory)/pack/ ./*.tgz
+                  npx lerna exec --no-private --no-sort --parallel -- mv -t $(Build.ArtifactStagingDirectory)/pack-scoped/ ./*.tgz
                 else
                   npm pack --unsafe-perm
-                  mv -t $(Build.ArtifactStagingDirectory)/pack/ ./*.tgz
+                  mv -t $(Build.ArtifactStagingDirectory)/pack-scoped/ ./*.tgz
                 fi
 
+          - ${{ if eq(variables.publishNonScopedPackages, true) }}:  
+            - ${{ each parameter in parameters.nonScopedPackages }}:
+              - task: Bash@3
+                displayName: Move Non-Scoped Package ${{parameter}}
+                inputs:
+                  targetType: 'inline'
+                  workingDirectory: ${{ parameters.buildDirectory }}
+                  script: |
+                    mv -t $(Build.ArtifactStagingDirectory)/pack-non-scoped/ $(Build.ArtifactStagingDirectory)/pack-scoped/${{parameter}}*.tgz
+
           - task: PublishBuildArtifacts@1
-            displayName: Publish Artifact - pack
+            displayName: Publish Artifact for Scoped Packages - pack-scoped
             inputs:
-              PathtoPublish: '$(Build.ArtifactStagingDirectory)/pack'
-              ArtifactName: 'pack'
+              PathtoPublish: '$(Build.ArtifactStagingDirectory)/pack-scoped'
+              ArtifactName: 'pack-scoped'
               publishLocation: 'Container'
 
+          - ${{ if eq(variables.publishNonScopedPackages, true) }}:  
```

Comment:
May be we can structure artifact directory as pack\scoped and pack\unscoped, so we can publish only one artifact?

Reply:
Updated to only publish one artifact, and use separate paths within it